### DISCUSSION TITLE
saving bokeh plots and display on dash

### DISCUSSION THREAD

**Discussion by GSav90** [kernc/backtesting.py#610] [Q&A] [answered]
First of all, thanks a lot for this awesome library. It is really a game changer.

I'm interested in saving the bokeh plot generated and display it on my dash app for different runs. So, I have couple of questions:

How can I save the bokeh plot and load it in my dash app?
I know the "backtest" object created has backtest.plot() object. How do I call this object from dash? Can you please guide a bit. I have never used bokeh and looking at the documents, I couldn't find anything concrete.
I see we have an option to create a file. Is it possible to load the html file or extract json from it?

**Comment by kernc** [comment, marked as answer]
When in doubt, just look in the source. [`Backtest.plot()`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest.plot) returns a [Bokeh Figure object](https://docs.bokeh.org/en/latest/docs/reference/plotting/helpers.html):
https://github.com/kernc/backtesting.py/blob/94d20da85e278102a0cc71b27a3a35b815e11648/backtesting/_plotting.py#L659-L668
which you can then load as HTML or maybe [as an image](https://docs.bokeh.org/en/latest/docs/reference/io.html). Best look to Dash documentation/community for how to do that?

**Reply by GSav90** [reply]
I can save the object as image/html, however loading bokeh plot in dash is something which I'm unable to find the answer to.

**Reply by kernc** [reply]
I think this is something better asked on a [Dash forum](https://community.plotly.com/c/python/25).

**Reply by GSav90** [reply]
Right. I'll post it there. Will add my answer here if I find it.

**Comment by zaferyumruk** [comment]
As you may know, dash is natively used with plotly. There are workarounds for what you ask but none of them seems perfect. 

If you insist on bokeh (as I do) I highly recommend you to consider switching panel or streamlit or write another plot function that uses plotly.
